src/error.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2019 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::ffi::*;
+use std::{ffi::CStr, os::raw::c_char};
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Extraction error: '{0}'")]
+    ExtractionError(String),
+
+    #[error("Io error: '{0}'")]
+    Io(#[from] std::io::Error),
+
+    #[error("ffi error: '{0}'")]
+    Ffi(#[from] std::ffi::NulError),
+
+    #[error("Utf error: '{0}'")]
+    Utf(#[from] std::str::Utf8Error),
+
+    #[error("Try from int error: '{0}'")]
+    TryInt(#[from] std::num::TryFromIntError),
+
+    #[error("Convert infallible error: '{0}'")]
+    ConvertInfallible(#[from] std::convert::Infallible),
+
+    #[error("Error to create the archive struct, is null")]
+    ArchiveNull,
+
+    #[error("The entry is null, failed to set the pathname")]
+    EntryNull,
+
+    #[error("File not found")]
+    FileNotFound,
+
+    #[error("Nix error: '{0}'")]
+    Nix(#[from] nix::Error),
+}
+
+pub(crate) fn handle_error(value: i32, archive: *mut archive) -> Result<()> {
+    let archive = &archive;
+    if value != ARCHIVE_OK {
+        return Err(unsafe { Error::from(archive_error_string(*archive)) });
```

Comment:
The `let` bind can  be dropped and the archive usage can just be passed without deref.